serverlessworkflow/modules/ROOT/pages/getting-started/create-your-first-workflow-service-with-kn-cli-and-vscode.adoc

```diff
@@ -0,0 +1,120 @@
+[source,bash]
+----
+cd ./my-sonataflow-project
+----
+* Open the folder in Visual Studo Code and examine the created `workflow.sw.json` using our extension.
+
+Now you can run the project and execute the workflow.
+
+[[proc-running-app-with-kn-cli]]
+== Running a Workflow project with Visual Studio Code and KN CLI
+
+Use the `run` command with kn workflow to build and run the {product_name} project in local development mode.
+
+* Run the project. This will start a `kogito-swf-devmode-nightly` image and map your local folder to this image.
+[source,bash]
+----
+kn workflow run
+----
+* The Development UI wil be accesible at `localhost:8080/q/dev`
+* You can now work on your project. Any changes will be picked up by the hot reload feature.
+* See xref:testing-and-troubleshooting/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-instances.adoc[Workflow instances] guide on how to run workflows via Development UI.
+* Once you are done developing your project navigate to the terminal that is running the `kn workflow run` command and hit `Ctlr+C` to stop the development environment.
+
+To deploy the finished project to a local cluster, proceed to next section.
+
+[[proc-deploying-app-with-kn-cli]]
+== Deploying a workflow project with Visual Studio Code and KN CLI
+
+Use the `deploy` command with kn workflow to deploy the {product_name} project into your local cluster.
+
+* Deploy to minikube
+[source,bash]
+----
+kn workflow deploy
+----
+* (Optional) Using k9scli you can examine your deployment.
+* In a different bash instance, create a port mapping:
+[source,bash]
+----
+minikube service hello --namespace default --url
+----
+* Use this URL to access your workflow instances using the Developer UI
+** <RETURNED_URL>/q/dev/org.kie.kogito.kogito-quarkus-serverless-workflow-devui/workflowInstances
+* To update the image run the `deploy` again, note that this may take some time.
+
+* To stop the deployment, use the `undeploy` command:
+[source,bash]
+----
+kn worklow undeploy
+----
+* You can validate your pod is terminating using k9s cli.
+
+[[proc-testing-application]]
+== Testing your workflow application
+
+To test your workflow application you can use any capable REST client out there. All that is needeed is the URL of your deployed worklow project.
+
+.Prerequisites
+* You have your workflow project deployed  using <<proc-deploying-app-with-kn-cli>> and you have the URL where it is deployed handy.
+
+.Testing your workflow application
+* To test your workflow project, access the Swagger UI on `<URL>/q/swagger-ui` to examine available endpoints.
+* In order to execute the workflow once, run: 
+[source,bash]
+----
+curl -X 'POST' \
+  '<URL>/hello' \
+  -H 'accept: */*' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "workflowdata": {}
+}'
+----
+* To examine executed instance you can use the GraphQL UI by navigating to
+`<URL>/q/graphl-ui` or calling additional REST queries.
```

Comment:
```suggestion
`<URL>/q/graphl-ui` or call additional REST queries.
```
Also, what additional REST queries you mean?

Reply:
I ll update,  wanted to give different options, I ll update